File: src/test/java/com/ebay/epd/dockerrunner/ContainerTest.scala
package com.ebay.epd.dockerrunner

import com.ebay.epd.dockerrunner.DockerRunner.Link
import org.scalatest.{Matchers, FlatSpec, FunSuite}

class ContainerTest extends FlatSpec with Matchers {

  import collection.JavaConversions._

  "Container" should "start and stop" in {
    val client = DockerHostFactory.dockerHostForEnvironment(System.getenv()).client
    val container = new Container(client, "spartans/docker-runner-image1", List[Link]())
    val initialNumberOfRunningContainers = client.listContainers().size()
    val startedContainer = container.start()
    val numberOfRunningContainersAfterStart = client.listContainers().size()
    startedContainer.stop()
    val finalNumberOfRunningContainers = client.listContainers().size()

    numberOfRunningContainersAfterStart should be(initialNumberOfRunningContainers + 1)
    finalNumberOfRunningContainers should be(initialNumberOfRunningContainers)
  }

  it should "only start once" in {
    val client = DockerHostFactory.dockerHostForEnvironment(System.getenv()).client
    val container = new Container(client, "spartans/docker-runner-image1", List[Link]())

    val startedContainer1 = container.start()
    val numberOfRunningContainersAfterFirstStart = client.listContainers().size()
    val startedContainer2 = container.start()
    val numberOfRunningContainersAfterSecondStart = client.listContainers().size()
    startedContainer1.stop()
    startedContainer2.stop()
    val finalNumberOfRunningContainers = client.listContainers().size()

    numberOfRunningContainersAfterFirstStart should be(numberOfRunningContainersAfterSecondStart)
    finalNumberOfRunningContainers should be(numberOfRunningContainersAfterSecondStart - 1)
    startedContainer1 should be(startedContainer2)
  }
}

File: src/test/java/com/ebay/epd/dockerrunner/ContainerTest.java
package com.ebay.epd.dockerrunner;

public class ContainerTest {

//    private final DockerHostFactory.DockerHost dockerHost = new DockerHostFactory().dockerHostForEnvironment(System.getenv());
//    private final DockerClient client = dockerHost.client();
//
//    @Test
//    public void shouldStartAndStopAContainer() throws Exception {
//        Container container = new Container(client, "commregistry-slc.corp.ebay.com/spartans/docker-runner-image1", Lists.<Link>newArrayList(), dockerHost.host(), Option.<String>None(), Option.<Memory>None(), Lists.<Container.Env>newArrayList(), Option.<String>None());
//        int initialNumberOfRunningContainers = client.listContainers().size();
//        StartedContainer startedContainer = container.start();
//        int numberOfRunningContainersAfterStart = client.listContainers().size();
//        startedContainer.stop();
//        int finalNumberOfRunningContainers = client.listContainers().size();
//        assertThat(numberOfRunningContainersAfterStart, is(initialNumberOfRunningContainers + 1));
//        assertThat(finalNumberOfRunningContainers, is(initialNumberOfRunningContainers));
//    }
//
//    @Test
//    public void shouldOnlyStartContainerOnce() throws Exception {
//        Container container = new Container(client, "commregistry-slc.corp.ebay.com/spartans/docker-runner-image1", Lists.<Link>newArrayList(), dockerHost.host(), Option.<String>None(), Option.<Memory>None(), Lists.<Container.Env>newArrayList(), Option.<String>None());
//        StartedContainer startedContainer1 = container.start();
//
//        int numberOfRunningContainersAfterFirstStart = client.listContainers().size();
//        StartedContainer startedContainer2 = container.start();
//        int numberOfRunningContainersAfterSecondStart = client.listContainers().size();
//        startedContainer1.stop();
//        startedContainer2.stop();
//        int finalNumberOfRunningContainers = client.listContainers().size();
//
//        assertThat(numberOfRunningContainersAfterFirstStart, is(numberOfRunningContainersAfterSecondStart));
//        assertThat(finalNumberOfRunningContainers, is(numberOfRunningContainersAfterSecondStart - 1));
//        assertThat(startedContainer1, is(startedContainer2));
//    }
}
